resolver-dns/src/main/java/io/netty/resolver/dns/DefaultDnsCache.java

```diff
@@ -171,18 +171,27 @@ private void cache0(DefaultDnsCacheEntry e, int ttl, EventLoop loop) {
             }
         }
 
-        scheduleCacheExpiration(entries, e, ttl, loop);
+        scheduleCacheExpiration(e, ttl, loop);
     }
 
-    private void scheduleCacheExpiration(final Entries entries,
-                                         final DefaultDnsCacheEntry e,
+    private void scheduleCacheExpiration(final DefaultDnsCacheEntry e,
                                          int ttl,
                                          EventLoop loop) {
         e.scheduleExpiration(loop, new Runnable() {
                     @Override
                     public void run() {
-                        if (entries.remove(e)) {
-                            resolveCache.remove(e.hostname);
+                        // We always remove all entries for a hostname once one entry expire. This is not the
+                        // most efficient to do but this way we can guarantee that if a DnsResolver
```

Comment:
what about keeping the cache purge behavior consistent, but triggering background queries as necessary (on demand, periodically, etc...) to refresh the expired entries? What is the DNS resolver's behavior in this circumstance?

Reply:
We may be able to optimize it but for now I think this fix is ok.... we may want to think more about how the whole cache is implemented and what informations are provided to it tho